src/pagination/internal.tsx

```diff
@@ -66,7 +81,18 @@ function PageButton({
 
 function PageNumber({ pageIndex, ...rest }: PageButtonProps) {
   return (
-    <PageButton className={styles['page-number']} pageIndex={pageIndex} {...rest}>
+    <PageButton
+      className={styles['page-number']}
+      pageIndex={pageIndex}
+      {...rest}
+      {...(rest.disabled
```

Comment:
It seems like double check is happening. First you skip propagating the metadata if `rest.disabled` and then you also check `disabled` prop inside the `PageButton` one more time.

This can be simplified

Reply:
The metadata that is being injected is different. The outer one adds only the position, the inner one the action and label

Reply:
oh well, now you can see how confusing this is. 
approved anyway